mask/eolearn/tests/test_utils.py

```diff
@@ -0,0 +1,79 @@
+import numpy as np
+import pytest
+
+from eolearn.mask.utils import ResizeLib, ResizeMethod, map_over_axis, spatially_resize_image
+
+
+@pytest.fixture(name="test_image", scope="module")
+def test_image_fixture():
+    """Image with a distinct value in each of the quadrants. In the bottom right quadrant there is a special layer."""
+    example = np.zeros((3, 100, 100, 4))
+    example[:, 50:, :50, :] = 1
+    example[:, :50, 50:, :] = 2
+    example[:, 50:, 50:, :] = 3
+    example[0, 50:, 50:, 0] = 10
+    example[0, 50:, 50:, 1] = 20
+    example[0, 50:, 50:, 2] = 30
+    example[0, 50:, 50:, 3] = 40
+    return example.astype(np.float32)
+
+
+def test_map_over_axis():
+    data = np.ones((5, 10, 10))
+    result = map_over_axis(data, lambda x: np.zeros((7, 20)), axis=0)
+    assert result.shape == (5, 7, 20)
+    result = map_over_axis(data, lambda x: np.zeros((7, 20)), axis=1)
+    assert result.shape == (7, 10, 20)
+    result = map_over_axis(data, lambda x: np.zeros((5, 10)), axis=1)
+    assert result.shape == (5, 10, 10)
+
+
+@pytest.mark.parametrize("method", ResizeMethod)
+@pytest.mark.parametrize("library", ResizeLib)
+@pytest.mark.parametrize("dtype", (np.float32, np.int32, bool))
+@pytest.mark.parametrize("new_size", [(50, 50), (35, 39), (271, 271)])
+def test_spatially_resize_image_new_size(method, library, dtype, new_size):
+    """Test that all methods and backends are able to downscale and upscale images of various dtypes."""
+    if library is ResizeLib.CV2:
+        if dtype == np.int32 and method is ResizeMethod.CUBIC or dtype == bool:
+            return
+
+    old_shape = (111, 111)
+    data_2d = np.arange(np.prod(old_shape)).astype(dtype).reshape(old_shape)
+    result = spatially_resize_image(data_2d, new_size, resize_method=method, resize_library=library)
+    assert result.shape == new_size
+
```

Comment:
Additionally you could also check that dtype is preserved.

Reply:
good call, added